### ISSUE TITLE
[REVIEW]: Capytaine: a Python-based linear potential flow solver

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1341]
**Submitting author:** @mancellin (<a href="http://orcid.org/0000-0002-0316-3230">Matthieu Ancellin</a>)
**Repository:** <a href="https://github.com/mancellin/capytaine" target ="_blank">https://github.com/mancellin/capytaine</a>
**Version:** v1.0.1
**Editor:** @leouieda
**Reviewer:** @mikaem, @harpolea
**Archive:** <a href="https://doi.org/10.5281/zenodo.2633744" target="_blank">10.5281/zenodo.2633744</a>

## Status

[![status](http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234/status.svg)](http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234"><img src="http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234/status.svg)](http://joss.theoj.org/papers/cd95bd8d080b463cbc39642ed283f234)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/mancellin/capytaine" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mikaem & @harpolea, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mancellin/capytaine">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.0.1
- [x] **Authorship:** Has the submitting author (@mancellin) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mancellin/capytaine">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.0.1
- [x] **Authorship:** Has the submitting author (@mancellin) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mikaem, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01341/joss.01341/10.21105.joss.01341.pdf)

**Comment by mikaem** [comment]
@leouieda This is a strong submission and I'm ready to sign off as it is. My only real issue, that I don't consider a blocker, is that the authors should put a little more effort into the installation. I think they would benefit greatly from using both conda-forge and pypi, and from making this code available on mac and windows, not just linux. I could build it myself on mac using conda-forge compilers, using instructions from [here](https://conda-forge.org/docs/maintainer/knowledge_base.html).

**Comment by leouieda** [comment]
@mikaem thank you for your comments. Would you mind opening an issue on the software repository to address these issues? I agree that it's a non-blocker if you managed to get installed but our goal is to help improve the software through the review. 

@mancellin would you like to try to fix these issues while we wait for @harpolea? I agree with @mikaem that using conda-forge is a better choice than using your own conda channel. I have a lot of experience with it and help set things up as well. I also agree that uploading to PyPI is a good thing as conda-forge often uses PyPI to get the source code for building packages.

**Comment by mikaem** [comment]
@leouieda just opened an [issue](https://github.com/mancellin/capytaine/issues/11) with a suggestion for recipe. @mancellin needs to take it from there.

**Comment by mancellin** [comment]
Yes, I agree that the distribution of the package should be improved.

I had not considered conda-forge, I will try it. Thank you @mikaem for the links and the recipe!

**Comment by mancellin** [comment]
Mac and Linux packages are now available on conda-forge.
I still have some issues building a Windows package. I'll try to keep myself informed on the conda(-forge) Fortran toolchain for Windows.

It is also on PyPI, although only as a source package.

**Comment by leouieda** [comment]
@mancellin thank you for keeping us updated :+1: 

:wave: @harpolea just checking in: I see that there are some checklist items remaining for you. Please let me know when you have a chance to go through those remaining items.

**Comment by harpolea** [comment]
@leouieda and @mancellin I'm really sorry for the delay - I was travelling last week and had far less time/wifi availability than I had anticipated. I have completed most of the checks I had intended to do, but just wanted to do a couple more things before I sign off on my review. I shall have a look at them this evening and should be finished by the end of the day.

**Comment by leouieda** [comment]
Thanks for the update, @harpolea :+1: I'm sure we're can all sympathize with your situation :slightly_smiling_face:

**Comment by harpolea** [comment]
Sorry again for the delay - I've finished up with the checks I had intended to do and recommend this for acceptance @leouieda 

This is a really nice package - the documentation in particular is very thorough and was great for a new user like myself!

**Comment by leouieda** [comment]
@harpolea thank you for your review!

**Comment by leouieda** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5802/smai-jcm.36 is OK
- 10.1115/OMAE2018-77924 is OK
- 10.1504/IJCSE.2009.029165 is OK
- 10.5334/jors.148 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.apor.2018.10.007 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01341/joss.01341/10.21105.joss.01341.pdf)

**Comment by leouieda** [comment]
@mancellin I have given the paper a final check and I'm happy to accept it for publication in JOSS :tada: :tada: 

There a few steps that we need to take before publication. Please make a new release with any changes that can about after submission (if required) and archive a copy of the source code on Zenodo or figshare. Once that is done, please post the DOI of the archive here.

**Comment by mancellin** [comment]
Thank you @leouieda, @harpolea and @mikaem!

I've read the paper once more and found a small typo in the bibliography. I have pushed a correction.

The DOI of the latest archive is 10.5281/zenodo.2633744

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01341/joss.01341/10.21105.joss.01341.pdf)

**Comment by leouieda** [comment]
@mancellin thanks for catching that!

**Comment by leouieda** [comment]
@whedon set v1.0.1 as version

**Comment by whedon** [comment]
OK. v1.0.1 is the version.

**Comment by leouieda** [comment]
@whedon set 10.5281/zenodo.2633744 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2633744" target="_blank">10.5281/zenodo.2633744</a> is the archive.

**Comment by leouieda** [comment]
:wave: @openjournals/joss-eics this submission is ready to be accepted

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/613

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/613, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5802/smai-jcm.36 is OK
- 10.1115/OMAE2018-77924 is OK
- 10.1504/IJCSE.2009.029165 is OK
- 10.5334/jors.148 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.apor.2018.10.007 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/614
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01341](https://doi.org/10.21105/joss.01341)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @mancellin, your JOSS paper is published!

Many thanks to our editor: @leouieda, and the reviewers: @mikaem, @harpolea — your contribution is greatly appreciated 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01341/status.svg)](https://doi.org/10.21105/joss.01341)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01341">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01341/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01341/status.svg
   :target: https://doi.org/10.21105/joss.01341
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01341/status.svg)](https://doi.org/10.21105/joss.01341)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss
